compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -296,23 +301,27 @@ BLangStatementExpression desugar(BLangQueryExpr queryExpr, SymbolEnv env,
         if (queryExpr.isStream) {
             resultType = streamRef.getBType();
         } else if (queryExpr.isTable) {
-            onConflictExpr = (onConflictExpr == null)
-                    ? ASTBuilderUtil.createLiteral(pos, symTable.nilType, Names.NIL_VALUE)
-                    : onConflictExpr;
             BLangVariableReference tableRef = addTableConstructor(queryExpr, queryBlock);
-            result = getStreamFunctionVariableRef(queryBlock,
-                    QUERY_ADD_TO_TABLE_FUNCTION, Lists.of(streamRef, tableRef, onConflictExpr, isReadonly), pos);
+            if (onConflictExpr == null) {
+                result = getStreamFunctionVariableRef(queryBlock,
+                        QUERY_ADD_TO_TABLE_FUNCTION, Lists.of(streamRef, tableRef, isReadonly), pos);
+            } else {
+                result = getStreamFunctionVariableRef(queryBlock,
+                        QUERY_ADD_TO_TABLE_FOR_ON_CONFLICT_FUNCTION, Lists.of(streamRef, tableRef, isReadonly), pos);
+            }
```

Comment:
Only the name changes between the two, right? I would use a ternary expression with `onConflictExpr == null` to get the function's Name assigned to a variable and use that in one usage of `getStreamFunctionVariableRef`.